src/quaternion.rs

```diff
@@ -173,6 +169,40 @@ impl<S: BaseFloat> Quaternion<S> {
             (self * scale1 + other * scale2) * Rad::sin(theta).recip()
         }
     }
+
+    /// Decompose a quaternion into a swing and a twist.
+    /// See [this stackoverflow question] and [this paper] for more information.
+    /// To get the corresponding swing do, for example,
+    ///
+    /// ```
+    /// # #[macro_use] extern crate cgmath;
+    /// # fn main() {
+    /// use std::f32::consts::PI;
+    /// use cgmath::{Quaternion, Euler, Vector3, InnerSpace, Deg};
+    /// // Rotation of 45 degrees about y axis then 45 degrees about x axis
+    /// let rotation = Quaternion::from(Euler { x: Deg(45.0), y: Deg(45.0), z: Deg(0.0)});
+    /// let twist = rotation.twist(Vector3::new(1.0, 1.0, 0.0).normalize(), None);
+    /// // We solve `swing * twist = rotation => swing = rotation & twist ^-1`.
+    /// let swing = rotation * twist.conjugate();
+    /// // if we recombine our rotations we get the original back.
+    /// assert_ulps_eq!(rotation, swing * twist);
+    /// panic!("swing: {:#?}, twist: {:#?}, rotation: {:#?}, new rotation: {:#?}",
+    /// Euler::from(swing), Euler::from(twist), Euler::from(rotation), Euler::from(swing * twist));
+    /// # }
+    /// ```
+    /// . There is a degenerate case where the direction is orthogonal to the direction of the
+    /// quaternion, and the angle is 180 degrees. In this case there are infinite possible
+    /// decompositions, so allow the user to specify one with `fallback`.
```

Comment:
I like this documentation! ❤️  - Wondering if it would be better just to return an `Option` in that case?

Reply:
Yeah could easily do that, if it's preferable. What do you think about the utility of the whole method?

Reply:
I think I'd like to leave this open until someone who understands this stuff has a look and says whether it's useful or not.